pkg/apis/caching/v1alpha1/image_types.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"reflect"
+	"sort"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/pkg/apis"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Image is a Knative abstraction that encapsulates the interface by which Knative
+// components express a desire to have a particular image cached.
+type Image struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state of the Image (from the client).
+	// +optional
+	Spec ImageSpec `json:"spec,omitempty"`
+
+	// Status communicates the observed state of the Image (from the controller).
+	// +optional
+	Status ImageStatus `json:"status,omitempty"`
+}
+
+// Check that Image can be validated and defaulted.
+var _ apis.Validatable = (*Image)(nil)
+var _ apis.Defaultable = (*Image)(nil)
+
+// ImageSpec holds the desired state of the Image (from the client).
+type ImageSpec struct {
+	Image string `json:"image"`
+
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
```

Comment:
What is this for, and how does it differ from ImagePullSecrets?

Reply:
This is to support [this](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account) authentication flow where the `imagePullSecrets` are indirectly specified via a K8s service account.